number_theory/modular.h

```diff
@@ -135,7 +135,7 @@ namespace modular_internal {
 
 // Tests if the type T is the Modular class.
 template <typename T>
-struct is_modular {
+struct ModularTypeImpl {
```

Comment:
I wonder if it is possible to make this struct a functor by overloading `operator()` as constexpr. Because, we basically use this struct as a `is_modular` function and we saw some examples from cppreference where they define `is_meowable` as constexpr function.

This would change `ModularTypeImpl<T>::value` to `is_modular<T>()` even though the example was simply using `is_meowable<T>` without the `operator()`.

What do you think?